component/src/main/java/org/wso2/carbon/identity/authenticator/LinkedIn/LinkedInAuthenticator.java

```diff
@@ -0,0 +1,350 @@
+     * values validations are not required.
+     *
+     * @param request  the http request
+     * @param response the http response
+     * @param context  the authentication context
+     * @throws AuthenticationFailedException
+     */
+    @Override
+    protected void initiateAuthenticationRequest(HttpServletRequest request,
+                                                 HttpServletResponse response, AuthenticationContext context)
+            throws AuthenticationFailedException {
+        try {
+            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+            if (authenticatorProperties != null) {
+                String clientId = authenticatorProperties.get(LinkedInAuthenticatorConstants.CLIENT_ID);
+                String authorizationEP = getAuthorizationServerEndpoint(authenticatorProperties);
+                if (authorizationEP == null) {
+                    authorizationEP = authenticatorProperties
+                            .get(LinkedInAuthenticatorConstants.OAUTH2_AUTHZ_URL);
+                }
+                String callbackurl = getCallbackUrl(authenticatorProperties);
+                String state = context.getContextIdentifier() + ","
+                        + LinkedInAuthenticatorConstants.LINKEDIN_LOGIN_TYPE;
+                state = getState(state, authenticatorProperties);
+                OAuthClientRequest authzRequest;
+                // This is the query string need to send getting email and
+                // profile
+                String queryString = LinkedInAuthenticatorConstants.QUERY_STRING;
+                authzRequest = OAuthClientRequest
+                        .authorizationLocation(authorizationEP)
+                        .setClientId(clientId)
+                        .setRedirectURI(callbackurl)
+                        .setResponseType(LinkedInAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE)
+                        .setState(state).buildQueryMessage();
+                String loginPage = authzRequest.getLocationUri();
+                String domain = request.getParameter("domain");
+                if (domain != null) {
+                    loginPage = loginPage + "&fidp=" + domain;
+                }
+                if (!queryString.startsWith("&")) {
+                    loginPage = loginPage + "&" + queryString;
+                } else {
+                    loginPage = loginPage + queryString;
+                }
+                response.sendRedirect(loginPage);
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("Error while retrieving properties. Authenticator Properties cannot be null");
+                }
+                throw new AuthenticationFailedException(
+                        "Error while retrieving properties. Authenticator Properties cannot be null");
+            }
+        } catch (IOException e) {
+            log.error("Exception while sending to the login page", e);
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-linkedIn/pull/3